cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfRobotComponent.cpp

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "UrdfRobotComponent.h"
+
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Unreal.h"
+#include "UrdfJointComponent.h"
+#include "UrdfParser.h"
+
+void UUrdfRobotComponent::initialize(UrdfRobotDesc* robot_desc)
+{
+    // initialize root link
+    UrdfLinkDesc* root_link_desc = robot_desc->root_link_desc_;
+    root_link_component_ = NewObject<UUrdfLinkComponent>(this, Unreal::toFName("UUrdfRobotComponent::link_component_" + root_link_desc->name_));
+    root_link_component_->initialize(root_link_desc);
+    link_components_[root_link_desc->name_] = root_link_component_;
+
+    // recursively build joint
+    ASSERT(root_link_desc->child_link_descs_.size() == root_link_desc->child_joint_descs_.size());
+    unsigned int num_child_links = root_link_desc->child_link_descs_.size();
+
+    for (unsigned int i = 0; i < num_child_links; i++) {
```

Comment:
This for loop is not needed, see my latest sketch.

Reply:
fixed. Added a `parent_joint_desc_` pointer for each `link_desc`